extension/pluginentityfactory.cpp

```diff
@@ -977,7 +977,18 @@ IServerNetworkable* CPluginEntityFactory::RecursiveCreate(const char* classname,
 			// If requested, attach a INextBot interface. The entity must be nextbotless and deriving from CBaseCombatCharacter
 			if (createNextBot && pEnt->MyCombatCharacterPointer() && !pEnt->MyNextBotPointer())
 			{
-				pEntityRecord->m_pNextBot = new ToolsNextBot(pEnt->MyCombatCharacterPointer());
+				INextBot* createdBot = nullptr;
+				IPluginFunction* factory = pFactory->GetNextBotFactory();
```

Comment:
`pFactory` will always be null at this point. You'll have to save the first factory pointer that requests for a NextBot attachment and use that instead, similar to how `pInitialActionFactory` is done.